packages/ckeditor5-dev-env/lib/release-tools/tasks/generatechangelog.js

```diff
@@ -7,59 +7,86 @@
 
 const conventionalChangelog = require( 'conventional-changelog' );
 const { tools, stream, logger } = require( '@ckeditor/ckeditor5-dev-utils' );
-const parserOpts = require( '../changelog/parser-options' );
-const writerOpts = require( '../changelog/writer-options' );
 const getNewReleaseType = require( '../utils/getnewreleasetype' );
-const utils = require( '../utils/changelog' );
+const hasCommitsFromLastRelease = require( '../utils/hascommitsfromlastrelease' );
+const cli = require( '../utils/cli' );
+const getPackageJson = require( '../utils/getpackagejson' );
 
 /**
  * Generates the release changelog based on commit messages in the repository.
  *
- * This method should be executed before the `tasks.createRelease` method.
+ * User can provide a version for the entry in changelog.
+ *
+ * If package does not have any commits, user has to confirm whether the changelog
+ * should be generated.
  *
  * @returns {Promise}
  */
-module.exports = function generateChangelog() {
+module.exports = function generateChangelog( newVersion = null ) {
 	const log = logger();
 
-	log.info( `Generating changelog for: ${ process.cwd() }` );
+	return new Promise( ( resolve ) => {
+		const packageJson = getPackageJson();
+
+		log.info( `Generating changelog for "${ packageJson.name }".` );
 
-	const shExecParams = { verbosity: 'warning' };
+		let promise = Promise.resolve();
 
-	return getNewReleaseType().then( ( response ) => {
-		// Bump the version for conventionalChangelog.
-		tools.shExec( `npm version ${ response.releaseType } --no-git-tag-version`, shExecParams );
+		if ( !newVersion ) {
+			promise = promise.then( () => getNewReleaseType() )
+				.then( ( response ) => {
+					const newReleaseType = ( hasCommitsFromLastRelease() ) ? response.releaseType : null;
 
-		return new Promise( ( resolve ) => {
-			// conventionalChangelog based on version in `package.json`.
-			conventionalChangelog( {}, null, { merges: undefined, firstParent: true }, parserOpts, writerOpts )
-				.pipe( saveChangelogPipe() );
+					return cli.provideVersion( packageJson.name, packageJson.version, newReleaseType );
+				} );
+		} else {
+			promise = promise.then( () => Promise.resolve( newVersion ) );
+		}
 
-			function saveChangelogPipe() {
-				return stream.noop( ( changes ) => {
-					let currentChangelog = utils.getChangelog();
+		return promise
+			.then( ( version ) => {
+				if ( version === 'skip' ) {
```

Comment:
If you type anything other than `skip`, provided input will be checking by the `semver` package. Most probably it will be rejected because your input will be marked as a invalid version.